iib/workers/tasks/build.py

```diff
@@ -260,6 +260,56 @@ def get_rebuilt_image_pull_spec(request_id):
     )
 
 
+def _get_resolved_bundles(bundles):
+    """
+    Get the pull specification of the bundle images using their digests.
+
+    Determine if the pull spec refers to a manifest list.
+    If so, simply use the digest of the amd64 arch in the manifest list.
+    If not a manifest list, it must be a v2s2 image manifest and should be used as it is.
+
+    :param list bundles: the list of bundle images to be resolved.
+    :return: the list of bundle images resolved to their digests.
+    :rtype: list
+    :raises IIBError: if unable to resolve the bundle image.
+    """
+    log.debug('Resolving bundles %s', ', '.join(bundles))
+    resolved_bundles = set()
+    for bundle_pull_spec in bundles:
+        skopeo_raw = skopeo_inspect(f'docker://{bundle_pull_spec}', '--raw')
+        if (
+            skopeo_raw.get('mediaType')
+            == 'application/vnd.docker.distribution.manifest.list.v2+json'
+        ):
+            arch_found = False
+            for manifest in skopeo_raw['manifests']:
+                if manifest['platform']['architecture'] == 'amd64':
+                    arch_found = True
+                    # The user may specify pull_spec with either tag or digest. Hence, make a call
+                    # to the _get_resolved_image to make sure we have the pull_spec with digest.
+                    resolved_pull_spec = _get_resolved_image(bundle_pull_spec)
+                    resolved_bundles.add(
+                        f'{resolved_pull_spec.rsplit("@", 1)[0]}@{manifest["digest"]}'
```

Comment:
What's wrong with using `resolved_pull_spec` directly?

Reply:
If it is a manifest list, and we use that digest (which resolved_pull_spec uses), opm is unable to resolve those to arch specific digest. Hence, we are providing it with the things it needs.

Reply:
Right. I misread the code.